digest.go

```diff
@@ -101,26 +101,21 @@ func FromString(s string) Digest {
 // error if not.
 func (d Digest) Validate() error {
 	s := string(d)
-
 	i := strings.Index(s, ":")
-
-	// validate i then run through regexp
-	if i < 0 || i+1 == len(s) || !DigestRegexpAnchored.MatchString(s) {
+	if i < 0 || i+1 == len(s) {
```

Comment:
The first condition should be `i <= 0` to catch the empty-algorithm-string case, which will let you drop the `algorithm == ""` block below.  Unit tests for `Digest("")`, `Digest(":")` (which we already have), `Digest(":foo")`, and `Digest("foo:")` (we already have `"sha256:"`) would ensure we'd handled all the empty-component cases.